sass/cd/cd-layout/_cd-layout.scss

```diff
@@ -35,15 +35,24 @@ body {
   }
 
   // Grid is used to fix the footer to the bottom of the screen, regardless of page min-height.
+  // Header height is fixed.
+  // Highlighted should contain Drupal messages and will be present even if empty.
+  // Main should grow to fit the available space.
+  // Soft footer is optional.
```

Comment:
The grid layout works if the soft footer is absent but it assumes the other 4 elements are present. I think this is a safe assumption as a default.
On pages that might have a hero (or whatever full viewport element) as an additional sibling, this will need to be adjusted.